dpnp/backend/kernels/dpnp_krnl_random.cpp

```diff
@@ -1082,19 +1081,18 @@ void dpnp_rng_standard_t_c(void* result, const _DataType df, const size_t size)
     if (dpnp_queue_is_cpu_c())
     {
         mkl_rng::gamma<_DataType> gamma_distribution(shape, d_zero, 1.0 / shape);
-        auto event_out = mkl_rng::generate(gamma_distribution, DPNP_RNG_ENGINE, size, result1);
-        event_out.wait();
-        event_out = mkl_vm::invsqrt(DPNP_QUEUE, size, result1, result1, no_deps, mkl_vm::mode::ha);
-        event_out.wait();
+        auto gamma_distr_event = mkl_rng::generate(gamma_distribution, DPNP_RNG_ENGINE, size, result1);
+
+        auto invsqrt_event = mkl_vm::invsqrt(DPNP_QUEUE, size, result1, result1, {gamma_distr_event}, mkl_vm::mode::ha);
 
-        sn = reinterpret_cast<_DataType*>(dpnp_memory_alloc_c(size * sizeof(_DataType)));
+        sn = reinterpret_cast<_DataType*>(malloc_shared(size * sizeof(_DataType), DPNP_QUEUE));
```

Comment:
Why we need direct "malloc" call here? It should be the same as "dpnp_memory_alloc_c" call

Reply:
I am using here pair: `malloc_shared` and `free` here. It does not change behavior.
But you're right, to avoid potentially undefined behavior, it's better to use our adopted allocator.